internal/model/transform.go

```diff
@@ -210,6 +216,45 @@ func (m *Model) InstantiateTemplates() error {
 	return nil
 }
 
+func (m *Model) instantiateField(pkgScope *ast.Package, field *ast.Field, templateTypes map[string]*ast.TypeReference) (*ast.Field, error) {
+	fieldCopy := *field
+	if len(field.Type.TemplateArguments) > 0 {
+		// the field itself is a templated compound type, and needs to
+		// be instantiated first.
+
+		// the template argument parameter must be recreated, as the
+		// field template arguments might differ from the struct template
+		// arguments. Example:
+		// struct<A, B, C> {
+		//    foo<int, C> field1;
+		//    bar<B, A> field2;
+		// }
+		newTemplateParameters := append([]*ast.TypeReference{}, field.Type.TemplateArguments...)
+		for ix, templateParameter := range newTemplateParameters {
+			if passedTemplateArgument, found := templateTypes[templateParameter.Name]; found {
+				newTemplateParameters[ix] = passedTemplateArgument
+			}
+		}
+
+		tt, err := m.instantiateType(pkgScope, field.Type, newTemplateParameters, "")
+		if err != nil {
+			return nil, err
+		}
+		fieldCopy.Type = tt
+		return &fieldCopy, nil
+	} else if tt := templateTypes[field.Type.Name]; tt != nil && field.Type.Package == "" {
```

Comment:
Consider following `if` statement guidelines from the effective go: https://go.dev/doc/effective_go#if as we can simplify things here.

What is more, if we have a map, consider using the `tt, ok := templateTypes[field.Type.Name]` pattern. This may still yield into a panic if there are `nil` items inside the map, but if that were the case, then we would know we have a much bigger problem.

Reply:
done